### PR TITLE
PFT-939 Add endpoints to upload/download job script files

### PR DESCRIPTION
**Pull request by fschuch**
#### What

* We need to add endpoints that can support the file editors until the Jobbergate Data Restructure project is complete.
* These endpoints should not be used to add additional files or to delete files that have been added.
* These will be temporary endpoints that should only be consumed by the frontend.

#### Why
Enhance usability on the UI.

`Task`: https://app.clickup.com/t/18022949/PFT-939

---

#### Peer Review
Please follow the upstream omnivector documentation concerning
[peer-review guidelines](https://github.com/omnivector-solutions/Documentation/blob/main/Contributing/pr_review_standards.md#peer-review).

### CODE REVIEW

File: jobbergate-api/jobbergate_api/apps/job_scripts/routers.py [hunk 1]
```diff
@@ -176,6 +181,76 @@ async def job_script_get(job_script_id: int = Query(...)):
     return response
 
 
+@router.patch(
+    "/job-scripts/{job_script_id}/upload",
+    status_code=status.HTTP_204_NO_CONTENT,
+    description="Endpoint to replace a job script file.",
+    dependencies=[Depends(guard.lockdown(Permissions.JOB_SCRIPTS_EDIT))],
+)
+async def job_script_replace_file_content(
```
Comment (dusktreader) [inline jobbergate-api/jobbergate_api/apps/job_scripts/routers.py:190]: This doesn't change the "last modified" field for a job script. Could be an issue later.

**Review by dusktreader — APPROVED** [review]

File: jobbergate-api/jobbergate_api/apps/job_scripts/routers.py [hunk 2]
```diff
@@ -176,6 +181,76 @@ async def job_script_get(job_script_id: int = Query(...)):
...
+
+    file_manager = JobScriptFiles.file_manager_factory(job_script_id)
+    file_content = job_script_file.file.read().decode("utf-8")
+
+    for s3_path in file_manager.keys():
+        root_dir = s3_path.parts[0]
+        if root_dir == JOBSCRIPTS_MAIN_FILE_FOLDER:
+            file_manager[s3_path] = file_content
+            logger.debug(f"Success replacing the main file from {job_script_id=}")
+            return None
```
Comment (matheushent) [inline jobbergate-api/jobbergate_api/apps/job_scripts/routers.py:214]: will this return 204??

Comment (fschuch) [reply jobbergate-api/jobbergate_api/apps/job_scripts/routers.py:214]: Yes, confirmed by the unit tests

Comment (matheushent) [reply jobbergate-api/jobbergate_api/apps/job_scripts/routers.py:214]: hmmm interesting, I usually use `Response(status=204)`. Nice to know.

File: jobbergate-api/jobbergate_api/apps/job_scripts/routers.py (hunk 1 above)
Comment (fschuch) [reply jobbergate-api/jobbergate_api/apps/job_scripts/routers.py:189]: solved in [2e55bb4](https://github.com/omnivector-solutions/jobbergate/pull/257/commits/2e55bb49b2eccea7a199557ef4346a6c1d772a0b)

**Pull request merged**